packages/datadog-instrumentations/src/child_process.js

```diff
@@ -0,0 +1,176 @@
+    return function () {
+      if (!childProcessChannelStart.hasSubscribers || arguments.length === 0) {
+        return childProcessMethod.apply(this, arguments)
+      }
+
+      const childProcessInfo = {
+        command: arguments[0],
+        args: arguments[1],
+        options: arguments[2]
+      }
+
+      if (Array.isArray(childProcessInfo.args)) {
+        childProcessInfo.command = childProcessInfo.command + ' ' + childProcessInfo.args.join(' ')
+      } else if (childProcessInfo.args == null) {
+        childProcessInfo.args = []
+      } else {
+        childProcessInfo.options = arguments[1]
+      }
+
+      if (shell ||
+        childProcessInfo?.options?.shell === true ||
+        typeof childProcessInfo?.options?.shell === 'string') {
+        childProcessInfo.shell = true
+      }
+
+      childProcessChannelStart.publish(childProcessInfo)
+
+      let error
+      let result
+      try {
+        result = childProcessMethod.apply(this, arguments)
+        return result
+      } catch (err) {
+        childProcessChannelError.publish(err)
+        error = err
+        throw err
+      } finally {
+        const exitCode = error?.status || error?.code || result?.status || 0
+        if (!error && (exitCode !== 0 || result?.error)) {
+          childProcessChannelError.publish()
+        }
+        childProcessChannelFinish.publish({ exitCode })
+      }
+    }
+  }
+}
+
+function wrapChildProcessCustomPromisifyMethod (customPromisifyMethod) {
+  return function () {
+    if (!childProcessChannelStart.hasSubscribers || arguments.length === 0) {
+      return customPromisifyMethod.apply(this, arguments)
+    }
+
+    const command = arguments[0]
+    childProcessChannelStart.publish({ command })
+
+    const promise = customPromisifyMethod.apply(this, arguments)
+    return promise.then((res) => {
+      childProcessChannelFinish.publish({ exitCode: 0 })
+
+      return res
+    }).catch((err) => {
+      childProcessChannelError.publish(err.status || err.code)
+      childProcessChannelFinish.publish({ exitCode: err.status || err.code || 0 })
+
+      return Promise.reject(err)
+    })
+  }
+}
+
+function wrapChildProcessAsyncMethod () {
+  return function wrapMethod (childProcessMethod) {
+    function wrappedChildProcessMethod () {
+      if (!childProcessChannelStart.hasSubscribers) {
+        return childProcessMethod.apply(this, arguments)
+      }
+
+      const childProcessInfo = {
+        command: arguments[0],
+        args: arguments[1],
+        options: arguments[2]
+      }
+
+      if (Array.isArray(childProcessInfo.args)) {
+        childProcessInfo.command = childProcessInfo.command + ' ' + childProcessInfo.args.join(' ')
+      } else if (childProcessInfo.args == null) {
+        childProcessInfo.args = []
+      } else {
+        childProcessInfo.options = arguments[1]
+      }
+
+      if (childProcessInfo?.options?.shell === true || typeof childProcessInfo?.options?.shell === 'string') {
```

Comment:
no point in having optional chaining for childProcessInfo, it can't possibly be empty

Reply:
fixed.